File: src/main/java/core/model/player/TrainerType.java
package core.model.player;

public enum TrainerType {
    /*
    0 = defensive, 1 = offensive, 2 = balanced
     */
    None(-1),
    Defensive(0),
    Offensive(1),
    Balanced(2);
    final int id;

    TrainerType(int i) {
        this.id = i;
    }

    public static  TrainerType fromInt(int i){
        return switch (i) {
            case 0 -> Defensive;
            case 1 -> Offensive;
            case 2 -> Balanced;
            default -> None;
        };
    }

    public static int toInt(TrainerType trainerTyp) {
        if ( trainerTyp != null) return trainerTyp.toInt();
        return None.id;
    }

    public int toInt(){
        return this.id;
    }
}

File: src/main/java/core/model/player/TrainerType.kt
package core.model.player

enum class TrainerType(val id: Int) {
    /*
    0 = defensive, 1 = offensive, 2 = balanced
     */
    None(-1),
    Defensive(0),
    Offensive(1),
    Balanced(2);

    fun toInt(): Int {
        return id
    }

    companion object {
        @JvmStatic
        fun fromInt(i: Int?): TrainerType {
            return if (i != null) {
                when (i) {
                    0 -> Defensive
                    1 -> Offensive
                    2 -> Balanced
                    else -> None
                }
            } else None
        }

        fun toInt(trainerTyp: TrainerType?): Int {
            return trainerTyp?.toInt() ?: None.id
        }
    }
}
